phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -262,7 +262,7 @@ public void preScannerOpen(org.apache.hadoop.hbase.coprocessor.ObserverContext<R
             // last possible moment. You need to swap the start/stop and make the
             // start exclusive and the stop inclusive.
             ScanUtil.setupReverseScan(scan);
-            if (scan.getFilter() != null && !(scan.getFilter() instanceof PagedFilter)) {
+            if (!(scan.getFilter() instanceof PagedFilter)) {
```

Comment:
Could you explain more about this change? Was there a previous bug where Scans without a filter weren't being paged correctly?

Reply:
Is there any problem with a PagedFilter having an internal Filter which is null?

Reply:
Previously, scans without filter were not being paged as we enabled them for filters only which were having large scans, but now after discussing with @kadirozde we are enabling them for all scans!